### PR TITLE
D8CORE-1261: Revised table styles.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Updates default styles for all tables

# Needed By (Date)
- At your leisure. 

# Urgency
- None

# Steps to Test

1. Review code
2. Click tugboat and review elements page for table styles

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
We may need a page of all html elements to review this.

**Comment by yvonnetangsu** [comment]
> We may need a page of all html elements to review this.

Let me know if you need anything from me.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
I'm open to ideas. Do we want a public page with all the HTML elements we have styled for both testing for us and for the public?

**Comment by yvonnetangsu** [comment]
> I'm open to ideas. Do we want a public page with all the HTML elements we have styled for both testing for us and for the public?

Yeah, either that or we separate things into child pages, e.g., one for all form elements. Actually, perhaps we make a list of all the HTML elements we want to include then we could decide if it's ok to put them all on one page or if that gets too long? 

Things I can think of:

- Tables
- Form stuff (text input, text area, radio/checkbox, select dropdown, submit buttons, whole form etc.)
- lists - default ul, ol, and dl, also our custom styles, e.g., Kerri's list with lines in between, big number lists

Anything else?

**Comment by sherakama** [comment]
Yeah, it might be a good idea to get them separated. We are often asked for form elements and should look into providing not only more styled options but components for them as well. Anyhow, my vote is separation.

**Comment by yvonnetangsu** [comment]
> Yeah, it might be a good idea to get them separated. We are often asked for form elements and should look into providing not only more styled options but components for them as well. Anyhow, my vote is separation.

Sounds good to me. I vote for separation too. I've been wanting to do/see some styling on our form elements also!

**Comment by sherakama** [comment]
If you'd like to take first pass at this please do as I don't think I will get the time to spend on it I would like to this month.

**Comment by sherakama** [comment]
FYI
@buttonwillowsix @jenbreese @kerri-augenstein

**Comment by kerri-augenstein** [comment]
@yvonnetangsu @sherakama, the Adapt project has specific needs for FORM ELEMENTS! YAY

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu @sherakama, the Adapt project has specific needs for FORM ELEMENTS! YAY! So we'll definitely be creating and executing them at the root of Decanter, and sub-theming off of those for SAA/OOD, within the next few months.

Cool - let me know if you have any designs for these, Kerri (both default and variants). I'd be happy to write myself a ticket, or @jenbreese and I could each take on some elements (just a suggestion and I totally know you guys might be super busy with D8).

**Comment by yvonnetangsu** [comment]
> > @yvonnetangsu @sherakama, the Adapt project has specific needs for FORM ELEMENTS! YAY! So we'll definitely be creating and executing them at the root of Decanter, and sub-theming off of those for SAA/OOD, within the next few months.
> 
> Cool - let me know if you have any designs for these, Kerri (both default and variants). I'd be happy to write myself a ticket, or @jenbreese and I could each take on some elements (just a suggestion and I totally know you guys might be super busy with D8).

@kerri-augenstein @buttonwillowsix

**Comment by kerri-augenstein** [comment]
@yvonnetangsu, hahaha, designs... hahahaha. Will be working on 'em in the next few weeks and getting design feedback from y'all.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by sherakama** [comment]
Putting up for review so it doesn't fall behind again.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-cvaxshmfudtyfzvce2yqzg0drd0fdmnr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f

Visual Diffs:
- https://dashboard.tugboat.qa/5e3deaa99366b12eb519bb6f/visualdiffs

**Comment by yvonnetangsu** [comment]
> Putting up for review so it doesn't fall behind again.

@sherakama  Just double checking - the table element isn't actually on the tugboat preview right?

**Comment by jenbreese** [comment]
@yvonnetangsu and @sherakama I couldn't find a table element either.

**Comment by sherakama** [comment]
@yvonnetangsu @jenbreese 

I totally thought we had an elements page somewhere. Let me add to this.

**Comment by sherakama** [comment]
Here is a live URL of this code in action: https://amptesting.sites.stanford.edu/wysiwyg/test-tables-layout-full-width

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu @jenbreese
> 
> I totally thought we had an elements page somewhere. Guess that page is gone. An alternative to test this branch would be to pull it into your current work and quickly set up a table.

Yeah, there's one on the previous version of the site, but not on this Netlify one.

Are we going with this design for the default table style? Big improvements over the current one (you fixed the double border issue). Just want to know if there's a design we're going for.

**Comment by kerri-augenstein** [comment]
@yvonnetangsu @jenbreese, yea, I think we should go with the design that's shown on the page Shea shared. 

https://amptesting.sites.stanford.edu/wysiwyg/test-tables-layout-full-width

We tweaked some things that I think work well in the root of Decanter.

File: core/src/scss/elements/table/_table.scss [hunk 1]
```diff
@@ -14,37 +14,41 @@
 // Style guide: Elements.Table
 //
 table {
-  @include margin(2em 0);
+  @include margin(2em 0 null 0);
```
Comment (yvonnetangsu) [inline core/src/scss/elements/table/_table.scss:17]: Should we remove margins outside of the element itself so we're consistent with all other Decanter components?

**Comment by yvonnetangsu** [comment]
Screen Shot 2020-04-30 at 9 35 48 AM
^ ^ ^ That's the margin-bottom I was referring to.

And have you seen this newer MARTY video btw? 😄 
https://www.youtube.com/watch?v=3x3SqeSdrAE

File: core/src/scss/elements/table/_table.scss (hunk 1 above)
Comment (jenbreese) [reply core/src/scss/elements/table/_table.scss:17]: I think consistency is a good principle to follow.

**Comment by jenbreese** [comment]
I like the idea of removing the margin-bottom. I think it would surprise users to have the extra space in a table.

**Comment by sherakama** [comment]
@yvonnetangsu @jenbreese ready for review again.

**Comment by jenbreese** [comment]
I think we want something more like this. But I don't think we want to use the * or to list out all the possibilities. 
```
&:last-child {
        p {
          @include margin(null null 0 null);
        }
      }
```

As it is in now, I get this look. (Borders I added to make easier to see.)
Screen Shot 2020-10-06 at 9 06 37 AM

File: core/src/scss/elements/table/_table.scss [hunk 2]
```diff
@@ -14,37 +14,45 @@
...
+  thead,
+  tbody {
 
-  thead {
     th,
     td {
-      background-color: $su-color-cloud;
+      @include padding(1.5rem);
+
+      &:last-child {
```
Comment (yvonnetangsu) [inline core/src/scss/elements/table/_table.scss:34]: Yeah...looks like this need a little fixup. Right now it's targeting the last `td` or `th` and setting the margin-bottom to 0, what we really want is to target the last element within each `td` or `th`.

**Comment by sherakama** [comment]
Ok, I changed it to target the child elements. I think we have to use `*` here as it could really be anything inside that td/th.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr617-yuceomezdiugb56vhokvohd6bauba4mf.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f7df8721e772d4c5715bd49

**Review by yvonnetangsu — APPROVED** [review]
Thanks for the fix! Looks good to me for base table style. GTG

**Comment by jenbreese** [comment]
@yvonnetangsu Want me to merge since you approved?

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu Want me to merge since you approved?

Hi @jenbreese Sure! Was waiting to see if you have anything else to add 😄  Thanks!

**Comment by sherakama** [comment]
Woot. 6 months of work. YAY!

**Comment by yvonnetangsu** [comment]
> Woot. 6 months of work. YAY!

You're doing better than me at un-staling 😂  🎉

**Pull request merged**